intercepting-filter/src/main/App.java

```diff
@@ -0,0 +1,245 @@
+import java.awt.*;
+import java.awt.event.*;
+public class App{
+	public static void main(String[] args){
+		FilterManager filterManager = new FilterManager(new Target());
+		filterManager.setFilter(new NameFilter());
+		filterManager.setFilter(new ContactFilter());
+		filterManager.setFilter(new AddressFilter());
+		filterManager.setFilter(new DepositFilter());
+		filterManager.setFilter(new OrderFilter());
+
+		Client client = new Client();
+		client.setFilterManager(filterManager);
+	}
+}
+
+interface Filter{
+	public String execute(String[] request);
+}
+
+class NameFilter implements Filter{
+	public String execute(String[] request){
+		if(request[0].equals("") || request[0].matches(".*[^\\w|\\s]+.*")){
+			return null;
+		}else return request[0];
+	}
+}
+
+class ContactFilter implements Filter{
+	public String execute(String[] request){
+		if(request[1].equals("") || request[1].matches(".*[^\\d]+.*") || request[1].length() != 11){
+			return null;
+		}else return request[1];
+	}
+}
+
+class AddressFilter implements Filter{
+	public String execute(String[] request){
+		if(request[2].equals("")){
+			return null;
+		}else return request[2];
+	}
+}
+
+class DepositFilter implements Filter{
+	public String execute(String[] request){
+		if(request[3].equals("")){
+			return null;
+		}else return request[3];
+	}
+}
+
+class OrderFilter implements Filter{
+	public String execute(String[] request){
+		if(request[4].equals("")){
+			return null;
+		}else return request[4];
+	}
+}
+
+class Target extends JFrame{
+	JTable jt;
+	JScrollPane jsp;
+	DefaultTableModel dtm;
+	JButton del;
+	public Target(){
+		super("Order System");
+		setDefaultCloseOperation(EXIT_ON_CLOSE);
+		setSize(640,480);
+		dtm = new DefaultTableModel(new Object[]{"Name", "Contact Number", "Address", "Deposit Number", "Order"},0);
+		jt = new JTable(dtm);
+		del = new JButton("Delete");
+		setup();
+	}
+	private void setup(){
+		setLayout(new BorderLayout());
+		JPanel bot = new JPanel();
+		add(jt.getTableHeader(), BorderLayout.NORTH);
+		bot.setLayout(new BorderLayout());
+		bot.add(del, BorderLayout.EAST);
+		add(bot, BorderLayout.SOUTH);
+		jsp = new JScrollPane(jt);
+		jsp.setPreferredSize(new Dimension(500,250));
+		add(jsp, BorderLayout.CENTER);
+
+		del.addActionListener(new DListener());
+
+		JRootPane rootPane = SwingUtilities.getRootPane(del); 
+		rootPane.setDefaultButton(del);
+		setVisible(true);
+	}
+	public void execute(String[] request){
+		//System.out.println(request[4]);
+		dtm.addRow(new Object[]{request[0],request[1],request[2],request[3],request[4]});
+	}
+
+	class DListener implements ActionListener{
+		@Override
+		public void actionPerformed(ActionEvent e){
+			int temp = jt.getSelectedRow();
+			if(temp == -1) return;
+			int temp2 = jt.getSelectedRowCount();
+			for(int i = 0; i < temp2; i++){
+				dtm.removeRow(temp);
+			}
+		}
+	}
+}
+
+class FilterChain{
+	private ArrayList<Filter> filters = new ArrayList<Filter>();
+	private Target target;
```

Comment:
Looks like FilterChain can't leave without Target.
You making assumption that setTarget() method will be called before call to execute().
I think constructor injection will do better here.